### PR TITLE
Comparing potential energy of bond

### PR DESCRIPTION
**Pull request by rishiloyola**

### CODE REVIEW

**Comment by ddamelin** [comment]
Here's some feedback on the coding:
- There is a lot of repetition of code for each model. That makes it hard to maintain, because you need to make changes in multiple places every time a change is needed. Can you think of a way to reduce duplicate code?
- When a number is a constant that is used in several places, it is better to set that value to some variable (or parameter as may be necessary in the interactive), and use the variable. This is similar to the code maintenance issue described in the previous bullet.
- There are some code indenting conventions that are not being followed.

Sam may have some other suggestions.

File: src/interactives/interactions/comparing-potential-energy-of-bond.json [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+                "var distx = x1-x2;",
+                "var disty = y1-y2;",
+                "var distance = Math.pow((distx*distx)+(disty*disty),0.5);",
+                "if(distance < 0.4)",
+                "set('countPotential', 5);",
+                "else",
+                "set('countPotential', 5+3*Math.pow(distance,0.5));",
+                "});",
+                "});"
+                ],
```
Comment (sfentress) [inline src/interactives/interactions/comparing-potential-energy-of-bond.json]: All scripts should be indented as if they were being written in a regular editor. You can see this, for example with the onLoad scripts in your mixing-liquids interactive. It makes it much easier to read.

File: src/interactives/interactions/comparing-potential-energy-of-bond.json [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+                "start();setPairwiseLJProperties(0, 2, { epsilon: 0 }); ",
+                "onDrag('atom',function(){",
+                "callEvery(25, function() {",
+                "var x1 = getAtomProperties(0).x;",
+                "var y1 = getAtomProperties(0).y;",
+                "var x2 = getAtomProperties(3).x;",
+                "var y2 = getAtomProperties(3).y;",
+                "var distx = x1-x2;",
+                "var disty = y1-y2;",
+                "var distance = Math.pow((distx*distx)+(disty*disty),0.5);",
```
Comment (sfentress) [inline src/interactives/interactions/comparing-potential-energy-of-bond.json]: In general, it's neater and easier to read if variable initializations follow the comma-separated list convention that you will have seen in other scripts (e.g. in mixing liquids):

```
var x1 = getAtomProperties(0).x,
    y1 = getAtomProperties(0).y,
    x2 = getAtomProperties(3).x,
...etc
```

This isn't a hard-and-fast rule, but when it's simply assigning a bunch of variables, it's easier to read.

**Comment by sfentress** [comment]
> There is a lot of repetition of code for each model. That makes it hard to maintain, because you need to make changes in multiple places every time a change is needed. Can you think of a way to reduce duplicate code?

Honestly, unless a feature such as the ability to use scripts from external files that [I suggested a few weeks ago](https://github.com/concord-consortium/lab/pull/90#issuecomment-108994731) gets implemented, there isn't really a practical way to "share" code between models.

The one way to do this would be to have a single model, instead of four different models. We have a number of models like this, for example "[non-bonding](http://lab.concord.org/interactives.html#interactives/interactions/non-bonding.json)" allows a user to change the elements, and it simply [updates the atom properties](https://github.com/concord-consortium/lab-interactives-site/blob/master/src/interactives/interactions/non-bonding.json#L170).

It's hard for me to judge at this point whether it would be worth re-writing this interactive to use one single model.

**Comment by ddamelin** [comment]
>Honestly, unless a feature such as the ability to use scripts from external files that I suggested a few weeks ago gets implemented, there isn't really a practical way to "share" code between models.

It looks like parameters are shared between models, so I was wondering if code to calculate might be tied to a model parameter. In each model there could be something like 

        onPropertyChange('time',function(time){", ....

in the onLoad function. Then that function could update some parameter (ex. updatePEFlag) that has an "onChange" attribute which contains the code for calculating the PE. 

@sfentress Do you think something like that would work?

**Comment by rishiloyola** [comment]
I did necessary changes in coding. You can now review it.

**Comment by ddamelin** [comment]
The behavior of the interactive seems pretty good now. I'm going to ask the Interactions team for feedback. However, I am noticing that the longer it runs the slower the dragging behavior is. (Or perhaps the more I drag the slower it gets.) I think this is because you have embedded a callEvery inside of an onDrag. Try removing the callEvery call. (This will also fix an indentation problem you have, as the callEvery function and its contents should be indented to show it is contained within the onDrag).

**Comment by ddamelin** [comment]
I made some additional whitespace changes. @rishiloyola please take a look for future interactives.

**Pull request merged**